flutter-idea/src/io/flutter/analytics/FlutterAnalysisServerListener.java

```diff
@@ -0,0 +1,521 @@
+        String filePath = file.getPath();
+        Instant nowInstant = Instant.now();
+        pathToErrorTimestamps.put(filePath, nowInstant);
+        pathToHighlightTimestamps.put(filePath, nowInstant);
+        pathToOutlineTimestamps.put(filePath, nowInstant);
+      }
+
+      @Override
+      public void selectionChanged(@NotNull FileEditorManagerEvent event) {
+      }
+
+      @Override
+      public void fileClosed(@NotNull final FileEditorManager source, @NotNull final VirtualFile file) {
+      }
+    };
+    project.getMessageBus().connect().subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, fileEditorManagerListener);
+    this.quickFixListener = new QuickFixListener();
+    this.requestListener = new FlutterRequestListener();
+    this.responseListener = new FlutterResponseListener();
+    DartAnalysisServerService analysisServer = DartAnalysisServerService.getInstance(project);
+    analysisServer.setServerLogSubscription(true);
+    analysisServer.addQuickFixListener(this.quickFixListener);
+    analysisServer.addRequestListener(this.requestListener);
+    analysisServer.addResponseListener(this.responseListener);
+  }
+
+  @NotNull
+  public static FlutterAnalysisServerListener getInstance(@NotNull final Project project) {
+    return Objects.requireNonNull(project.getService(FlutterAnalysisServerListener.class));
+  }
+
+  @NotNull
+  private static String safelyGetString(JsonObject jsonObject, String memberName) {
+    if (jsonObject != null && StringUtil.isNotEmpty(memberName)) {
+      JsonElement jsonElement = jsonObject.get(memberName);
+      if (jsonElement != null) {
+        return Objects.requireNonNull(jsonElement.getAsString());
+      }
+    }
+    return "";
+  }
+
+  @Override
+  public void dispose() {
+    // This is deprecated and marked for removal in 2021.3. If it is removed we will
+    // have to do some funny stuff to support older versions of Android Studio.
+    LookupManager.getInstance(project).removePropertyChangeListener(this::onPropertyChange);
+  }
+
+  @Override
+  public void computedAnalyzedFiles(List<String> list) {
+  }
+
+  @Override
+  public void computedAvailableSuggestions(@NotNull List<AvailableSuggestionSet> list, int[] ints) {
+  }
+
+  @Override
+  public void computedCompletion(String completionId,
+                                 int replacementOffset,
+                                 int replacementLength,
+                                 List<CompletionSuggestion> completionSuggestions,
+                                 List<IncludedSuggestionSet> includedSuggestionSets,
+                                 List<String> includedElementKinds,
+                                 List<IncludedSuggestionRelevanceTag> includedSuggestionRelevanceTags,
+                                 boolean isLast,
+                                 String libraryFilePathSD) {
+  }
+
+  @Override
+  public void computedErrors(String path, List<AnalysisError> list) {
```

Comment:
is this getting called for every file in a project?
If it is called for more than each root path, I'm concerned we are logging too much.

Reply:
Not every file, but all Dart files plus some unexpected files. All three `AndroidManifest.xml` files were checked plus `pubspec.yaml`.